test/sync/src/commonTest/kotlin/io/realm/kotlin.test/Utils.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.test
+
+import kotlin.reflect.KClass
+import kotlin.test.assertFailsWith
+
+// NOTE: Copy from :base:commonTest. It is unclear if there is an easy way to share test code like
```

Comment:
The pattern is to put it in `:base:commonMain` as `sync` depends on `base`.

Reply:
I tried that, but the code I'm trying to share is only in `:base:commonTest` and it doesn't seem like the test configurations can be shared like that :/

Reply:
But stuff like `TestHelper` in `base/src/commonMain` is used in ex. `SyncedRealmTest` 🤔

Reply:
Yes, `commonMain` works fine...but things in `commonTest` don't and our custom asserts methods require the test dependency.

Reply:
But `commonMain` of the `base` is sort or our _common test_-code ... why can't we just put it in that?

Reply:
Or is it too dangerous to have the test dependencies for `main` 🤔

Reply:
Maybe just leave it and the we can pool it with the "reconsider module structure"-task

Reply:
Yeah, not 100% sure what the best approach is. I'll add it to our DevEx list for now